### PR TITLE
Endpoint Management APIs

### PR DESCRIPTION
**Pull request by pzl**
Work in progress of the API requests

Previews: 
- [Get Endpoint ](https://security-docs_1239.docs-preview.app.elstc.co/guide/en/security/master/get-endpoint-api.html)
- [List Endpoints ](https://security-docs_1239.docs-preview.app.elstc.co/guide/en/security/master/list-endpoints-api.html)
- [Isolate a host](https://security-docs_1239.docs-preview.app.elstc.co/guide/en/security/master/host-isolation-api.html)
- [Release a host](https://security-docs_1239.docs-preview.app.elstc.co/guide/en/security/master/host-isolation-release-api.html)

### CODE REVIEW

File: docs/management/api/list-endpoints-api.asciidoc [hunk 1]
```diff
@@ -1,11 +1,313 @@
...
+
+[source,sh]
+--------------------------------------------------
+POST /api/endpoint/metadata
+{
+	"paging_properties": [
+		{ "page_size": 10 },
+		{ "page_index": 0 }
+	]
+}
```
Comment (jmikell821) [inline docs/management/api/list-endpoints-api.asciidoc:75]: @pzl this API call is the same is `List Endpoints`. Doesn't appear to have any of the filter properties. Is this correct?

Comment (pzl) [reply docs/management/api/list-endpoints-api.asciidoc:75]: Correct, this is List without filters. We should probably add another example with the filters used

File: docs/management/api/host-isolation-api.asciidoc [hunk 2]
```diff
@@ -1,11 +1,96 @@
 [[host-isolation-api]]
 === Isolate a host
 
-Enter API description here.
+Isolates a host running {endpoint-sec} from the network.
 
-NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Console].
+NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Run Elasticsearch API requests].
+
+`Isolated` is a persistent status until the endpoint is given a release command. You must have `superuser` role to perform this action, and at least a Platinum license level.
```
Comment (joepeeples) [inline docs/management/api/host-isolation-api.asciidoc:8]:
```suggestion
`Isolated` is a persistent status until the endpoint is given a release command. You must have the `superuser` role to perform this action, and at least a Platinum license level.
```

File: docs/management/api/host-isolation-api.asciidoc [hunk 3]
```diff
@@ -1,11 +1,96 @@
...
 
 ==== Request URL
 
+`POST <kibana host>:<port>/api/endpoint/isolate`
 
 ==== Request body
+
+a JSON object with these fields:
```
Comment (joepeeples) [inline docs/management/api/host-isolation-api.asciidoc:16]:
```suggestion
A JSON object with these fields:
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+You must have `superuser` role to perform this action. It is available to all license levels.
+
+==== Request URL
+
+`POST <kibana host>:<port>/api/endpoint/unisolate`
+
+==== Request body
+
+a JSON object with these fields:
```
Comment (joepeeples) [inline docs/management/api/host-isolation-release-api.asciidoc:16]:
```suggestion
A JSON object with these fields:
```

File: docs/management/api/list-endpoints-api.asciidoc [hunk 5]
```diff
@@ -1,11 +1,332 @@
...
+NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Run Elasticsearch API requests].
 
 ==== Request URL
 
+`POST <kibana host>:<port>/api/endpoint/metadata`
+
 
 ==== Request body
+
+a JSON object with these fields:
```
Comment (joepeeples) [inline docs/management/api/list-endpoints-api.asciidoc:15]:
```suggestion
A JSON object with these fields:
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 6]
```diff
@@ -0,0 +1,98 @@
+[[host-isolation-release-api]]
+=== Release an isolated Host
```
Comment (joepeeples) [inline docs/management/api/host-isolation-release-api.asciidoc:2]:
```suggestion
=== Release an isolated host
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+Removes a host's isolation status and allows it to rejoin a network.
+
+NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Run Elasticsearch API requests].
+
+You must have `superuser` role to perform this action. It is available to all license levels.
```
Comment (joepeeples) [inline docs/management/api/host-isolation-release-api.asciidoc:8]:
```suggestion
You must have the `superuser` role to perform this action. It is available to all license levels.
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+==== Request body
+
+a JSON object with these fields:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`endpoint_ids` |Array (String) |The IDs of each endpoint you wish to release |Yes
```
Comment (joepeeples) [inline docs/management/api/host-isolation-release-api.asciidoc:22]:
```suggestion
|`endpoint_ids` |Array (String) |The IDs of each endpoint you wish to release. |Yes
```

**Review by joepeeples — COMMENTED** [review]
A few small comments, otherwise looks good!

File: docs/management/api/host-isolation-api.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-api.asciidoc:8]:
```suggestion
`Isolated` is a persistent status until the endpoint is given a release command. You must have the `superuser` role to perform this action and at least a Platinum license.
```

File: docs/management/api/host-isolation-api.asciidoc [hunk 9]
```diff
@@ -1,11 +1,96 @@
...
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`endpoint_ids` |Array (String) |The IDs of each endpoint you wish to isolate |Yes
+|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here, and the isolated event will be logged in any cases associated with those specified alerts. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-api.asciidoc:23]:
```suggestion
|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here. The isolated event will be logged in cases associated with the specified alerts. |No
```

File: docs/management/api/host-isolation-api.asciidoc [hunk 10]
```diff
@@ -1,11 +1,96 @@
...
 
 ==== Request body
+
+a JSON object with these fields:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`endpoint_ids` |Array (String) |The IDs of each endpoint you wish to isolate |Yes
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-api.asciidoc:22]:
```suggestion
|`endpoint_ids` |Array (String) |The IDs of each endpoint you want to isolate. |Yes
```

File: docs/management/api/host-isolation-api.asciidoc [hunk 11]
```diff
@@ -1,11 +1,96 @@
...
+|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here, and the isolated event will be logged in any cases associated with those specified alerts. |No
+|`case_ids` |Array (String) |Logs the action taken on any specified cases. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-api.asciidoc:24]:
```suggestion
|`case_ids` |Array (String) |Logs the action taken on specified cases. |No
```

File: docs/management/api/host-isolation-api.asciidoc [hunk 12]
```diff
@@ -1,11 +1,96 @@
...
+|`comment` |String |Attach a comment to this action's log. The comment text will appear in any associated cases. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-api.asciidoc:25]:
```suggestion
|`comment` |String |Attach a comment to this action's log. The comment text will appear in associated cases. |No
```

File: docs/management/api/get-endpoint-api.asciidoc [hunk 13]
```diff
@@ -1,11 +1,141 @@
 [[get-endpoint-api]]
 === Get endpoint
 
-Enter API description here.
+Retrieves metadata about a single machine running {endpoint-sec}.
 
-NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Console].
+NOTE: Console supports only Elasticsearch APIs. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Run Elasticsearch API requests].
```
Comment (nastasha-solomon) [inline docs/management/api/get-endpoint-api.asciidoc:6]:
```suggestion
NOTE: Console supports Elasticsearch APIs only. Console doesn't allow interactions with {kib} APIs. You must use `curl` or another HTTP tool instead. For more information, refer to https://www.elastic.co/guide/en/kibana/current/console-kibana.html[Run Elasticsearch API requests].
```

Comment (nastasha-solomon) [reply docs/management/api/get-endpoint-api.asciidoc:6]: Just wanted to clarify that this note is saying that Console will only support Elasticsearch APIs and no other APIs.

File: docs/management/api/host-isolation-release-api.asciidoc (hunk 7 above)
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-release-api.asciidoc:8]:
```suggestion
You must have the `superuser` role to perform this action. It is available to all license levels.
```

File: docs/management/api/host-isolation-release-api.asciidoc (hunk 8 above)
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-release-api.asciidoc:22]:
```suggestion
|`endpoint_ids` |Array (String) |The IDs of each endpoint you want to release. |Yes
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here, and the released event will be logged in any cases associated with those specified alerts. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-release-api.asciidoc:23]:
```suggestion
|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here. The released event will be logged in cases associated with the specified alerts. |No
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+|`case_ids` |Array (String) |Logs the action taken on any specified cases. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-release-api.asciidoc:24]:
```suggestion
|`case_ids` |Array (String) |Logs the action taken on specified cases. |No
```

File: docs/management/api/host-isolation-release-api.asciidoc [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+|`comment` |String |Attaches a comment to this action's log. The comment text will appear in any associated cases. |No
```
Comment (nastasha-solomon) [inline docs/management/api/host-isolation-release-api.asciidoc:25]:
```suggestion
|`comment` |String |Attaches a comment to this action's log. The comment text will appear in associated cases. |No
```

File: docs/management/api/list-endpoints-api.asciidoc [hunk 17]
```diff
@@ -1,11 +1,332 @@
...
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`paging_properties` |Array a|A set of page instruction Objects that contain the following:
+
+* `page_size` (number): size of results per page. 1 to 10000
```
Comment (nastasha-solomon) [inline docs/management/api/list-endpoints-api.asciidoc:23]:
```suggestion
* `page_size` (number): Size of results per page. 1 to 10000
```

File: docs/management/api/list-endpoints-api.asciidoc [hunk 18]
```diff
@@ -1,11 +1,332 @@
...
+* `page_index` (number): the zero-based page of results to retrieve
```
Comment (nastasha-solomon) [inline docs/management/api/list-endpoints-api.asciidoc:24]:
```suggestion
* `page_index` (number): The zero-based page of results to retrieve.
```

File: docs/management/api/list-endpoints-api.asciidoc [hunk 19]
```diff
@@ -1,11 +1,332 @@
...
+
+A sample value for `paging_properties` would be:
+
+`paging_properties: [{page_size: 20}, {page_index: 0}]`
+
+ |No
+|`filters` |Object a|Filters to apply to the list query. It may contain the following filters as properties:
+
+* `kql` (string): a KQL string
```
Comment (nastasha-solomon) [inline docs/management/api/list-endpoints-api.asciidoc:33]:
```suggestion
* `kql` (string): A KQL string.
```

File: docs/management/api/list-endpoints-api.asciidoc [hunk 20]
```diff
@@ -1,11 +1,332 @@
...
+* `host_status` (array, string): a set of agent health statuses to filter by.
```
Comment (nastasha-solomon) [inline docs/management/api/list-endpoints-api.asciidoc:34]:
```suggestion
* `host_status` (array, string): A set of agent health statuses to filter by.
```

**Review by nastasha-solomon — APPROVED** [review]
Made some minor recs. LGTM otherwise!

**Review by benironside — APPROVED** [review]

**Pull request merged**